tests/vanilla/utils/types.test.tsx

```diff
@@ -1,3 +1,5 @@
+/* eslint @typescript-eslint/no-unused-vars: off */
```

Comment:
@JoshuaKGoldberg is there an option to mark `typeof foo` as used?

Reply:
Not at the moment, no. But I think this is an interesting use case that _might_ be enough to warrant one?

We'd talked about it before in:
* https://github.com/typescript-eslint/typescript-eslint/issues/9679
* https://github.com/typescript-eslint/typescript-eslint/issues/9697

...but neither is _exactly_ what's happening here. 

For this specific file, I think the user need here is that you...

* Need to have the typings tests code wrapped in functions so that they doesn't actually run
* Are using `function Component`s for that, which are getting flagged as unused

...is that right?

If so: I'd suggest naming them something like `function _test` per the `varsIgnorePattern`. That way you can keep `no-unused-vars` enabled & have it just not care about the functions.

Example benefit: `types.test.tsx` has an unused import (`expect`) right now that would be caught by the rule. 🧹

Reply:
Hm, that's unfortunate.

I was talking about `typeof foo`, not about `function test`, which I already resolved.

Thanks for all of your help. I will go with overrides for tests.

Reply:
Oh! Sorry, I misread. But great!